subsystem/kubeapi_test.go

```diff
@@ -881,73 +894,75 @@ var _ = Describe("[kube-api]cluster installation", func() {
-			host := registerNode(ctx, *infraEnv.ID, hostname, ips[i])
-			hosts = append(hosts, host)
-		}
-		for _, host := range hosts {
-			checkAgentCondition(ctx, host.ID.String(), v1beta1.ValidatedCondition, v1beta1.ValidationsFailingReason)
-			hostkey := types.NamespacedName{
-				Namespace: Options.Namespace,
-				Name:      host.ID.String(),
-			}
-			agent := getAgentCRD(ctx, kubeClient, hostkey)
-			Expect(agent.Status.ValidationsInfo).ToNot(BeNil())
-		}
-		generateFullMeshConnectivity(ctx, ips[0], hosts...)
-		for _, h := range hosts {
-			generateDomainResolution(ctx, h, clusterDeploymentSpec.ClusterName, "hive.example.com")
-			generateCommonDomainReply(ctx, h, clusterDeploymentSpec.ClusterName, clusterDeploymentSpec.BaseDomain)
-		}
+		ignitionTokenSecretName := "ignition-token"
+		ignitionEndpointToken := "abc"
+		deploySecret(ctx, kubeClient, ignitionTokenSecretName, map[string]string{common.IgnitionTokenKeyInSecret: ignitionEndpointToken})
 
-		By("Approve Agents")
-		for _, host := range hosts {
-			hostkey := types.NamespacedName{
-				Namespace: Options.Namespace,
-				Name:      host.ID.String(),
-			}
-			Eventually(func() error {
-				agent := getAgentCRD(ctx, kubeClient, hostkey)
-				agent.Spec.Approved = true
-				return kubeClient.Update(ctx, agent)
-			}, "30s", "10s").Should(BeNil())
+		ip := hostutil.GenerateIPv4Addresses(1, defaultCIDRv4)[0]
+		host := registerNode(ctx, *infraEnv.ID, "host1", ip)
+		hostkey := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      host.ID.String(),
 		}
-		installkey := types.NamespacedName{
+		clusterkey := types.NamespacedName{
 			Namespace: Options.Namespace,
 			Name:      clusterDeploymentSpec.ClusterInstallRef.Name,
 		}
-		By("Verify ClusterDeployment ReadyForInstallation")
-		checkAgentClusterInstallCondition(ctx, installkey, hiveext.ClusterRequirementsMetCondition, hiveext.ClusterAlreadyInstallingReason)
-		By("Delete ClusterDeployment")
-		err := kubeClient.Delete(ctx, getClusterDeploymentCRD(ctx, kubeClient, clusterKey))
+		agent := getAgentCRD(ctx, kubeClient, hostkey)
+		agent.Spec.IgnitionEndpointTokenReference = &v1beta1.IgnitionEndpointTokenReference{
+			Namespace: Options.Namespace,
+			Name:      ignitionTokenSecretName,
+		}
+		err := kubeClient.Update(ctx, agent)
 		Expect(err).To(BeNil())
-		By("Verify AgentClusterInstall was deleted")
+
+		i := 0
 		Eventually(func() bool {
-			aci := &hiveext.AgentClusterInstall{}
-			err := kubeClient.Get(ctx, installkey, aci)
-			return apierrors.IsNotFound(err)
-		}, "30s", "10s").Should(Equal(true))
-		By("Verify ClusterDeployment Agents were deleted")
-		Eventually(func() int {
-			return len(getClusterDeploymentAgents(ctx, kubeClient, clusterKey).Items)
-		}, "2m", "2s").Should(Equal(0))
+			dbCluster, err := common.GetClusterFromDBWhere(db, common.UseEagerLoading, common.IncludeDeletedRecords, "kube_key_name = ? and kube_key_namespace = ?", clusterkey.Name, clusterkey.Namespace)
```

Comment:
`clusterkey` isn't defined in this `It` scope
That's probably why the test fails to find the cluster in the DB...

Reply:
clusterkey := types.NamespacedName{
			Namespace: Options.Namespace,
			Name:      clusterDeploymentSpec.ClusterInstallRef.Name,
		}